### DISCUSSION TITLE
LiteLLM V2?

### DISCUSSION THREAD

**Discussion by ghost** [BerriAI/litellm#4383] [General]
Are the recent pydantic v1 -> v2 change https://github.com/BerriAI/litellm/pull/3670 + the upcoming ModelResponseChunk for streaming fix https://github.com/BerriAI/litellm/pull/4219 considered breaking changes? 

As such, would it make sense to release the upcoming fix under a v2 with a migration guide?

**Comment by ghost** [comment]
@andersskog @yujonglee thoughts on this?

**Comment by yujonglee** [comment]
@krrishdholakia   👋

I understand we now have additional `ModelResponseChunk` class to represent some data correctly, but I guess the actual data is same right? I don't think it will break anyone's code.

In my understanding, major version bump is for public API change.

For ex, 
- if we remove `litellm.completion` and add `litellm.cpl`, than it is breaking change.
- if we decided to accept model with `md` field rather than `model` field, than it is breaking change. 

But introducing new type to fix the linting error? I don't think it is change that need major version bump.

**Reply by ghost** [reply]
ok, that's good to understand. 

Yes - there's no change in format, just the pydantic object. I was worried in case somebody had asserts in their streaming logic 

```python
for chunk in response: 
   assert(chunk, ModelResponse) 
```

^ this would need to change.

**Reply by ghost** [reply]
When it's merged in, i can make it a minor bump (1.40.x -> 1.41.0) 

and add a migration guide on docs + release notes. 

Anything else you'd add to this? @yujonglee

**Reply by yujonglee** [reply]
I think minor bump with migration guide is enough!
